enhancements/cluster-logging/workload-identity-auth-for-azure-monitor-logs.md

```diff
@@ -0,0 +1,295 @@
+  - Include workload identity authentication.
+  - Include long-lived credentials through `client secrets`.
+
+### Workflow Description
+
+The Vector collector will do the following for workload identity authentication:
+
+1. Determine the authentication type to be workload identity when a token path is provided.
+2. Retrieve the OpenShift service account token from the local volume.
+3. Exchange the OpenShift token with Microsoft identity platform for a short-lived access token.
+4. Use the access token in the log forwarding request to Azure Monitor Logs.
+
+The Vector collector will do the following for credential secret authentication:
+
+1. Determine the authentication type to be client secret when a client secret is provided.
+2. Request an access token from the Microsoft identity platform using the client secret.
+3. Use the access token in the log forwarding request to Azure Monitor Logs.
+
+The ClusterLogForwarder will:
+
+1. Conditionally project the service account token if workload identity authentication is configured.
+2. Create the collector configuration with the required fields for the Log Ingestion API, including either the path to the projected service account token or the client secret.
+
+### Proposed API
+
+#### New `AzureLogIngestion` output type
+
+Output
+
+```Go
+// New output type of AzureLogIngestion which can live side by side with the AzureMonitor output
+type AzureLogIngestion struct {
+  // The Data Collection Endpoint (DCE) or Data Collection Rule (DCR) logs ingestion endpoint URL.
+  //
+  // https://learn.microsoft.com/en-us/azure/azure-monitor/logs/logs-ingestion-api-overview#endpoint
+  //
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:XValidation:rule="isURL(self)", message="invalid URL"
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Url",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  URL string `json:"url"`
+
+  // The Data Collection Rule's (DCR) Immutable ID
+  //
+  // A unique identifier for the data collection rule. This property and its value are automatically created when the DCR is created.
+  //
+  // https://learn.microsoft.com/en-us/azure/azure-monitor/data-collection/data-collection-rule-structure#properties
+  //
+  // +kubebuilder:validation:Required
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="DCR Immutable ID",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  DcrImmutableId string `json:"dcrImmutableId"`
+
+  // The stream in the Data Collection Rule (DCR) that should handle the custom data
+  // 
+  // https://learn.microsoft.com/en-us/azure/azure-monitor/data-collection/data-collection-rule-structure#input-streams
+  //
+  // +kubebuilder:validation:Required
+  // +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Stream Name",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+  StreamName string `json:"streamName"`
```

Comment:
Should this allow templating?  This sounds like it is the "tenant" which could be a dynamic value?  Maybe there is a concern with "grouping" here and the way batches of records are submitted that would make it not feasible to support a template...

Reply:
You could template this but it wouldn't be ideal. The stream isn't something that is created on the fly. You can think of the `streamName` as an existing table within Azure. It must be defined and exist in order for logs to be able to flow into it. 

If we template this, users must ensure the table exists within Azure otherwise they wouldn't get the logs. 

So if they want to template `{.log_type}-{.namespace_name}-my-azure-stream`, they must ensure all possible renditions of `log_type` and `namespace_name`, depending on what they want to collect, exist within Azure.

e.g `application-ns1-my-azure-stream`, `application-ns2-my-azure-stream`, `infrastructure-default-my-azure-stream`,... etc.

Reply:
This can be a future enhancement if needed.  I wonder if we need a validation to restrict templating?  I guess it depends what precedence we have

Reply:
I don't think we need validation. We can just tell users that templating is not supported. I also think it depends on upstream Vector's sink and their configurable fields if it allows templating. If it doesn't it would just throw an error.